ts/Dashboards/EditMode/Menu/MenuItem.ts

```diff
@@ -168,8 +170,9 @@ class MenuItem {
                     id: options.id,
                     name: options.id,
                     title: options.text || '',
-                    items: options.items || [],
-                    value: options.value || ''
+                    items: options.items as SelectFormFieldItem[] || [],
+                    value: options.value || '',
+                    onchange: options.events && options.events.change
```

Comment:
In case of using `options.events && options.events.change` in each IF, you can return it to the const and use here or replace with `(options.events || {}).change`. Up to you ;)